pkg/devices/rdma.go

```diff
@@ -18,22 +18,65 @@
 package devices
 
 import (
+	"github.com/golang/glog"
 	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
 
 	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
 	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
 )
 
 type rdmaSpec struct {
-	isSupportRdma bool
-	deviceSpec    []*pluginapi.DeviceSpec
+	deviceID   string
+	deviceType types.DeviceType
 }
 
-func newRdmaSpec(rdmaResources []string) types.RdmaSpec {
+// NewRdmaSpec returns the RdmaSpec
+func NewRdmaSpec(dt types.DeviceType, id string) types.RdmaSpec {
+	if dt == types.AcceleratorType {
+		return nil
+	}
+	return &rdmaSpec{deviceID: id, deviceType: dt}
+}
+
+func (r *rdmaSpec) IsRdma() bool {
+	if len(r.getRdmaResources()) > 0 {
+		return true
+	}
+	var bus string
+	//nolint: exhaustive
+	switch r.deviceType {
+	case types.NetDeviceType:
+		bus = "pci"
+	case types.AuxNetDeviceType:
+		bus = "auxiliary"
+	default:
+		return false
+	}
+	// Checking for netlink param for exclusive RDMA use case
```

Comment:
can you add more information here why we need to check netlink param in this case. (requested by sebastian)